Fitness/Controllers/FilterViewController.swift

```diff
@@ -746,3 +533,224 @@ class FilterViewController: UIViewController, UICollectionViewDelegateFlowLayout
         setupConstraints()
     }
 }
+
+//MARK: CollectionViewDelegateFlowLayout
+extension FilterViewController: UICollectionViewDelegateFlowLayout {
+    func collectionView(_ collectionView: UICollectionView, didDeselectItemAt indexPath: IndexPath) {
+        for i in 0..<selectedGyms.count{
+            if (selectedGyms[i] == gyms[indexPath.row].id){
+                selectedGyms.remove(at: i)
+                return
+            }
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        selectedGyms.append(gyms[indexPath.row].id)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        return CGSize(width: 35 + gyms[indexPath.row].name.count*10, height: 31)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return gyms.count
+    }
+
+}
+
+//MARK: CollectionViewDataSource
+extension FilterViewController: UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: GymFilterCell.identifier, for: indexPath) as! GymFilterCell
+        
+        cell.gymNameLabel.text = gyms[indexPath.row].name
+        cell.gymNameLabel.sizeToFit()
+        
+        if selectedGyms.contains(gyms[indexPath.row].id){
+            collectionView.selectItem(at: indexPath, animated: true, scrollPosition: .right)
+            cell.isSelected = true
+        }
+        return cell
+    }
+}
+
+//MARK: TableViewDataSource
+extension FilterViewController: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
```

Comment:
Note for later, this method should probably get split up. Complicated currently. Make note please